src/ApacheOrcDotNet.OptimizedReader/OrcReader.cs

```diff
@@ -209,14 +207,24 @@ public RowIndex GetColumnRowIndex(int columnId, int stripeId)
                     && s.ColumnId == columnId
                 ).Single();
 
-                var decompressedData = _byteRangeProvider.DecompressByteRange(
-                     rowIndexStream.FileOffset,
-                     rowIndexStream.Length,
-                     _fileTail.PostScript.Compression,
-                     (int)_fileTail.PostScript.CompressionBlockSize
-                 );
+                var compressedBuffer = ArrayPool<byte>.Shared.Rent(rowIndexStream.Length);
+                var decompressedBuffer = ArrayPool<byte>.Shared.Rent(rowIndexStream.Length * 1032);
```

Comment:
These buffers should all be using the same MaxBufferSize included in the footer...  But they may contain more than one chunk...  So I guess that new auto-sized buffer mechanism.

Reply:
I've updated these to use the same buffer length as used for data reads. I added 20ms to the overall execution time (~10%) due to the buffer length. But in the other hand, it is now configurable as well.

Reply:
Why do you think it added so much time?

Reply:
We now use a different config for metadata decompression, which dropped the time back between 80 an 90 ms.

The increase was due to the Rent/Return of big buffers when we were loading streams and row groups for caching.